dotCMS/src/main/java/com/dotcms/mock/request/FakeHttpServletRequest.java

```diff
@@ -0,0 +1,32 @@
+package com.dotcms.mock.request;
```

Comment:
We do not need this class. We already have a complete mocked request and response.
```
        HttpServletRequest requestProxy =
            new MockAttributeRequest(
                new MockSessionRequest(
                    new MockHttpRequest(host.getHostname(), uri).request()
                ).request()
            ).request();
        HttpServletResponse responseProxy = new BaseResponse().response();

        responseProxy.setContentType("text/html");
        requestProxy.setAttribute("User-Agent", userAgent);
        requestProxy.setAttribute("idInode", ident.getId());
        requestProxy.setAttribute(WebKeys.HTMLPAGE_LANGUAGE, Long.toString(viewingLang));
        requestProxy.getSession().setAttribute(com.dotmarketing.util.WebKeys.REDIRECT_AFTER_LOGIN, uri);
                    responseProxy.sendError(401, "The requested page/file is unauthorized");
```

 See for usage: 
https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java#L723

Reply:
Done